src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributes.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class ReadOnlyTagHelperAttributes<TAttributeType> : IReadOnlyList<TAttributeType>
```

Comment:
@Eilon yes you can. Getting it to buidl isn't the problem; doing things like:

```C#
foreach (var attribute in output.Attributes)
{
    // attribute is of type object.
}
```

or

```C#
var attribute = output.Attributes.FirstOrDefault(); // attribute is object
```

where it has to resolve the type ends up falling back to the base enumerable which results in the resolved type being `object` unless you specify otherwise.